session/src/main/scala/akka/stream/alpakka/cassandra/scaladsl/CassandraFlow.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.cassandra.scaladsl
+
+import akka.NotUsed
+import akka.cassandra.session.scaladsl.{ CassandraSession, CassandraSessionRegistry }
+import akka.cassandra.session.{ CassandraSessionSettings, CassandraWriteSettings }
+import akka.dispatch.ExecutionContexts
+import akka.stream.scaladsl.Flow
+import com.datastax.oss.driver.api.core.cql.{ BatchStatement, BatchType, BoundStatement, PreparedStatement }
+
+import scala.collection.JavaConverters._
+import scala.concurrent.{ ExecutionContext, Future }
+
+/**
+ * Scala API to create Cassandra flows.
+ */
+class CassandraFlow {
+
+  /**
+   *
+   * @param sessionSettings
+   * @param writeSettings
+   * @param stmt CQL statement
+   * @param statementBinder
+   * @tparam T
+   * @return
+   */
+  def createWithPassThrough[T](
+      sessionSettings: CassandraSessionSettings,
+      writeSettings: CassandraWriteSettings,
+      stmt: String,
+      statementBinder: (T, PreparedStatement) => BoundStatement): Flow[T, T, NotUsed] =
+    Flow
+      .setup { (materializer, _) =>
+        implicit val executionContext: ExecutionContext = materializer.system.dispatcher
+        val session = CassandraSessionRegistry(materializer.system).sessionFor(sessionSettings, executionContext)
+        val prepareStatement: Future[T => Future[T]] =
+          session.prepare(stmt).map(s => executePrepared(session, s, statementBinder)(_))
+        Flow[T].mapAsync(writeSettings.parallelism) { t =>
+          prepareStatement.flatMap(executeWriteForElement => executeWriteForElement(t))
+        }
+      }
+      .mapMaterializedValue(_ => NotUsed)
+
+  /**
+   *
+   * @param session
+   * @param writeSettings
+   * @param stmt
+   * @param statementBinder
+   * @tparam T
+   * @return
+   */
+  def createWithPassThrough[T](
+      session: CassandraSession,
+      writeSettings: CassandraWriteSettings,
+      stmt: PreparedStatement,
+      statementBinder: (T, PreparedStatement) => BoundStatement): Flow[T, T, NotUsed] = {
+    val executeWriteForElement: T => Future[T] = executePrepared(session, stmt, statementBinder)
+    Flow[T].mapAsync(writeSettings.parallelism)(t => executeWriteForElement(t))
```

Comment:
One scary thing with `parallelism > 1` is that the writes can be performed in different order from the stream order, since there are several possible async hops in the future composition. Worth mentioning in docs, and recommend 1 if order is important.